### ISSUE TITLE
Uncaught TypeError: Cannot read property 'skip' of undefined

### ISSUE THREAD

**Issue by nicksergeant** [t9md/atom-narrow#276]
**Steps to reproduce:**

1. Run `narrow:search`
2. Hit the Up Arrow

I'm new to Narrow so I'm not sure if the expected behavior is to cycle through previous searches (that's sorta what I was expecting), but it throws this error instead.

**Atom**: 1.24.0 x64
**Electron**: 1.6.16
**OS**: Mac OS X 10.12.6
**Thrown From**: [narrow](https://github.com/t9md/atom-narrow) package 0.57.1

### Stack Trace

Uncaught TypeError: Cannot read property 'skip' of undefined

```
At /Users/nsergeant/.atom/packages/narrow/lib/items.js:143

TypeError: Cannot read property 'skip' of undefined
    at findItem.item (/packages/narrow/lib/items.js:143:53)
    at Items.findItem (/packages/narrow/lib/items.js:124:11)
    at Items.findPromptOrNormalItem (/packages/narrow/lib/items.js:143:17)
    at editor.onDidChangeCursorPosition.event (/packages/narrow/lib/narrow-editor.js:165:31)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at TextEditor.cursorMoved (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2701:31)
    at Selection.markerDidChange (/Applications/Atom.app/Contents/Resources/app/src/selection.js:924:25)
    at Selection.marker.onDidChange.e (/Applications/Atom.app/Contents/Resources/app/src/selection.js:21:45)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at DisplayMarker.module.exports.DisplayMarker.notifyObservers (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:248:33)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:53:32
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at Marker.module.exports.Marker.emitChangeEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:428:26)
    at Marker.module.exports.Marker.update (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:369:18)
    at Marker.module.exports.Marker.setHeadPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:144:25)
    at DisplayMarker.module.exports.DisplayMarker.setHeadBufferPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:146:38)
    at changePosition (/Applications/Atom.app/Contents/Resources/app/src/cursor.js:86:25)
    at Cursor.changePosition (/Applications/Atom.app/Contents/Resources/app/src/cursor.js:713:11)
    at Cursor.setBufferPosition (/Applications/Atom.app/Contents/Resources/app/src/cursor.js:85:16)
    at setBufferRow (/packages/narrow/lib/utils.js:102:10)
    at NarrowEditor.moveUpOrDownWrap (/packages/narrow/lib/narrow-editor.js:255:7)
    at /packages/narrow/lib/ui.js:90:50)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:388:36)
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:621:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:412:28)
    at WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:108:40)
```

### Commands

```
     -0:03.4.0 narrow:search (input.hidden-input)
     -0:03.3.0 vim-mode-plus:activate-insert-mode (atom-text-editor.editor.narrow.narrow-editor.search.vim-mode-plus.normal-mode.prompt)
     -0:02.8.0 intentions:highlight (input.hidden-input)
     -0:02.8.0 core:move-up (input.hidden-input)
     -0:02.7.0 intentions:highlight (input.hidden-input)
```

### Non-Core Packages

```
busy-signal 1.4.3 
cursor-history 0.13.0 
django-templates 0.5.0 
git-blame 1.5.0 
git-time-machine 1.5.9 
hyperclick 0.1.5 
intentions 1.1.5 
jump-to-import 3.6.5 
language-haml 0.25.2 
linter 2.2.0 
linter-eslint 8.4.1 
linter-ui-default 1.6.10 
narrow 0.57.1 
persistent-undo 1.3.0 
prettier-atom 0.50.0 
sync-settings 0.8.3 
vim-mode-plus 1.28.0 
```

**Comment by t9md** [comment]
Could reproduce, thank you for reporting again.
Will fix soon.

**Comment by nicksergeant** [comment]
Cool! I’ve been a little busy this week but I’m interested in diving into Narrow’s code. Love what you’ve done here. Thanks for making it.

**Comment by t9md** [comment]
Thanks!
narrow is super important achievement for me!
It's not just searching tool.
Its code navigation tool to help refactor/investigate/reading big codes.
I cannot imagine using editor which doesn't have narrow equivalent feature.

**Comment by t9md** [comment]
Released fix and history improvement as v0.58.0
For history feature, see #277 or CHANGELOG.md
